internal/comment/handler.go

```diff
@@ -114,107 +138,176 @@ func (h *CommentHandler) matchingComments(ctx context.Context) ([]Comment, error
+	if err != nil {
+		return PostResult{Posted: false}, err
+	}
 
-	latestMatchingComment, err := h.LatestMatchingComment(ctx)
+	matchingComments, err := h.matchingComments(ctx)
 	if err != nil {
-		return false, err
+		return PostResult{Posted: false}, err
 	}
 
-	if latestMatchingComment != nil {
-		if latestMatchingComment.Body() == bodyWithTag {
-			log.Infof("Not updating comment since the latest one matches exactly: %s", color.HiBlueString(latestMatchingComment.Ref()))
-			return false, nil
+	if len(matchingComments) > 0 {
+		latestMatchingComment := matchingComments[len(matchingComments)-1]
+
+		latestValidAt := latestMatchingComment.ValidAt()
+		if validAt != nil && latestValidAt != nil && validAt.Before(*latestValidAt) {
+			msg := fmt.Sprintf("Not updating comment since the latest one is newer: %s", latestMatchingComment.Ref())
+			log.Warning(msg)
+			return PostResult{Posted: false, SkipReason: msg}, nil
+		}
+
+		if latestMatchingComment.Body() == bodyWithTags {
+			msg := fmt.Sprintf("Not updating comment since the latest one matches exactly: %s", latestMatchingComment.Ref())
+			log.Info(msg)
+			return PostResult{Posted: false, SkipReason: msg}, nil
 		}
 
-		log.Infof("Updating comment %s", color.HiBlueString(latestMatchingComment.Ref()))
+		log.Infof("Updating comment %s", latestMatchingComment.Ref())
 
-		err := h.PlatformHandler.CallUpdateComment(ctx, latestMatchingComment, bodyWithTag)
+		err := h.PlatformHandler.CallUpdateComment(ctx, latestMatchingComment, bodyWithTags)
 		if err != nil {
-			return false, h.newPlatformError(err)
+			return PostResult{Posted: false}, h.newPlatformError(err)
 		}
 	} else {
 		if skipNoDiff {
-			log.Infof("Not creating initial comment since there is no resource or cost difference")
-			return false, nil
+			msg := "Not creating initial comment since there is no resource or cost difference"
+			log.Info(msg)
+			return PostResult{Posted: false, SkipReason: msg}, nil
 		}
 
 		log.Info("Creating new comment")
 
-		comment, err := h.PlatformHandler.CallCreateComment(ctx, bodyWithTag)
+		comment, err := h.PlatformHandler.CallCreateComment(ctx, bodyWithTags)
 		if err != nil {
-			return false, h.newPlatformError(err)
+			return PostResult{Posted: false}, h.newPlatformError(err)
 		}
 
-		log.Infof("Created new comment %s", color.HiBlueString(comment.Ref()))
+		log.Infof("Created new comment %s", comment.Ref())
 	}
 
-	return true, nil
+	return PostResult{Posted: true}, nil
 }
 
 // NewComment creates a new comment with the given body.
-func (h *CommentHandler) NewComment(ctx context.Context, body string) error {
-	bodyWithTag := h.PlatformHandler.AddMarkdownTag(body, h.Tag)
+func (h *CommentHandler) NewComment(ctx context.Context, body string, opts *CommentOpts) error {
+	var validAt *time.Time
+	if opts != nil {
+		validAt = opts.ValidAt
+
+		if opts.SkipNoDiff {
+			log.Warning("SkipNoDiff option is not supported for new comments")
```

Comment:
`logging.Logger` please, and let's change all the rest. Thanks. (note, this isn't me just being a pain, it should hopefully make migration away from logrus easier in the future)